common/src/main/java/io/homeassistant/companion/android/common/data/integration/Entity.kt

```diff
@@ -649,6 +649,15 @@ fun <T> Entity<T>.friendlyState(context: Context, options: EntityRegistryOptions
             }
         }
     }
+
+    if (appendUnitOfMeasurement) {
+        val unit = (attributes as? Map<*, *>)?.get("unit_of_measurement")?.toString()
```

Comment:
Is this safe? Inspired by other usages of it in the `Entity` class.

If not, how to do it?

Reply:
Yes this is fine, the app doesn't have a typesafe alternative and as long as you're using known values from the map it shouldn't cause any issues.